src/main/scala/au/com/titanclass/streams/telemetry/StreamMetricExporter.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 Titan Class Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package au.com.titanclass.streams.telemetry
+import java.util
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.{ Attributes, OverflowStrategy }
+import akka.stream.scaladsl.{ BroadcastHub, Keep, Source }
+import io.opentelemetry.sdk.metrics.`export`.MetricExporter
+import io.opentelemetry.sdk.metrics.data.MetricData
+
+object StreamMetricExporter {
+  def apply(bufferSize: Int)(implicit system: ActorSystem): StreamMetricExporter =
+    new StreamMetricExporter(bufferSize)
+}
+
+/**
+  * Provides a source of metrics
+  */
+class StreamMetricExporter(bufferSize: Int)(implicit system: ActorSystem) extends MetricExporter {
+
+  private val (metricQueue, metricSource) = Source
+    .queue[MetricData](bufferSize, OverflowStrategy.dropHead.withLogLevel(Attributes.LogLevels.Off))
+    .toMat(BroadcastHub.sink(1))(Keep.both)
+    .run()
+
+  /**
+    * Returns a source of metrics
+    */
+  def source: Source[MetricData, NotUsed] =
+    metricSource
+
+  override def `export`(metrics: util.Collection[MetricData]): MetricExporter.ResultCode = {
+    metrics.forEach { metric =>
+      val _ = metricQueue.offer(metric)
```

Comment:
Do you need to assign this val for linting purposes? looks like it could otherwise be shortened to 

`metrics.forEach(metricQueue.offer)`

Reply:
It’s a wart to not use an assignment where a value is returned. It could indicate a mistake, hence the wart remover plugin highlighting this.

Reply:
Fair enough, I suppose it's a consequence of not caring about the return value of the Java API in this case. Perhaps worth writing an extension method wrapping this call with a Scala idiomatic Unit return type assuming the return value is really unimportant and if this was to be called in many places in the code but not worth it for one instance!